ethereum/miniheader/miniheader.go

```diff
@@ -0,0 +1,21 @@
+package miniheader
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// MiniHeader is a representation of a succinct Ethereum block headers
+type MiniHeader struct {
+	Hash   common.Hash
+	Parent common.Hash
+	Number *big.Int
+	Logs   []types.Log
+}
+
+// ID returns the MiniHeader's ID
+func (m *MiniHeader) ID() []byte {
```

Comment:
Hmm.. `ID` doesn't have anything to do with Ethereum. It's something we made up for compatibility with the `db` package.

Instead of defining this here, could we include create a wrapper type with the `ID` method in the `meshdb` package? Something like:

```go
type MiniHeader ethereum.MiniHeader

func (m *MiniHeader) ID() []byte {
    return m.Hash.Bytes()
}
```

Then whenever we save a mini header in the database, we would just need to cast it to `meshdb.MiniHeader`.

Reply:
I first tried to do that but it required 4 casts (not only when saving) and was generally very cumbersome. I've instead put a `// HACK` comment above it and if you feel strongly about it, I'm happy to discuss this further.